app/src/main/java/com/android/unio/ui/event/EventDetails.kt

```diff
@@ -110,11 +110,22 @@ fun EventScreen(
 
   val isSaved = user!!.savedEvents.contains(event!!.uid)
 
+  // Save button should be extracted to a utils in the future
   val onClickSaveButton = {
     if (isSaved) {
       user!!.savedEvents.remove(event!!.uid)
+      val newEvent = event!!.copy(numberOfSaved = event!!.numberOfSaved - 1)
+      eventViewModel.updateEventWithoutImage(
+          newEvent,
+          onSuccess = {},
+          onFailure = { e -> Log.e("EventCard", "Failed to update event: $e") })
     } else {
       user!!.savedEvents.add(event!!.uid)
+      val newEvent = event!!.copy(numberOfSaved = event!!.numberOfSaved + 1)
+      eventViewModel.updateEventWithoutImage(
+          newEvent,
+          onSuccess = {},
+          onFailure = { e -> Log.e("EventCard", "Failed to update event: $e") })
```

Comment:
Testing this mechanism in the UI test shouldn't be very difficult and would put your line coverage above 80% for sure. Do you think you could do that? Since it's pretty much the feature that you implemented, I do think it would deserve a test.

Reply:
I agree that it isn't very difficult but it feels absurd to add a whole UI test for something that should be refactored ASAP, but if you don't agree I'll get on it right now

Reply:
That would have been the plan initially but we have to see if we're going to have time to develop that with the time remaining. But what I meant by this needs to be refactored is specifically the onClickSaveButton function which is not complete here in EventDetails: it doesn't use the appropriate UserViewModel methods and does not register the notification handler, both of which are out of the scope of this PR and will require the function to be rewritten

Reply:
Ah yes you had mentioned it in your last comment, that's fine then 👍